File: src/gui-tesing-library/Components/GTProcess.cs
namespace gui_tesing_library.Components;

public class GTProcess : IGTProcess
{
    private readonly GTProcess_CS gTProcess_CS;

    public GTProcess(GTProcess_CS gTProcess_CS)
    {
        this.gTProcess_CS = gTProcess_CS;
    }

    public string Name => gTProcess_CS.GetName();
    public bool IsAlive => gTProcess_CS.IsAlive();


    public long GetRamUsage()
    {
        return gTProcess_CS.GetRamUsage();
    }

    public void kill()
    {
        gTProcess_CS.kill();
    }
}

File: src/GtLibrary/GTProcess.cpp
#include "pch.h"
#include "GTProcess.h"
#include <windows.h>
#include <stdexcept>
#include "SystemCallsFactory.h"
std::wstring GTProcess::GetName() const {
    return SystemCallsFactory::GetSystemCalls()->GetProcesName(handle);
}

bool GTProcess::IsAlive() const {
	return SystemCallsFactory::GetSystemCalls()->IsProcesAlive(handle);
}

std::vector<GTWindow> GTProcess::GetWindowsOfProcess() const {
	std::vector<HWND> handles = SystemCallsFactory::GetSystemCalls()->GetWindowsOfProcess(this->handle);
	std::vector<GTWindow> windows;
	for (HWND handle : handles) {
		GTWindow window = GTWindow(handle);
	
			windows.push_back(window);
		
	}
	return windows;
}


long GTProcess::GetRamUsage() const {
return	SystemCallsFactory::GetSystemCalls()->GetRamUsageOfProcess(this->handle);
}

void GTProcess::Kill() {
    SystemCallsFactory::GetSystemCalls()->KillProcess(this->handle);
}

HANDLE GTProcess::GetHandle() {
	return this->handle;
}